lib/Backend/BackendUtils.cpp

```diff
@@ -34,7 +34,7 @@ static llvm::cl::OptionCategory BackendUtilsCat("Glow Backend Utils Options");
 
 static llvm::cl::opt<bool> reuseActivationsMemory(
     "reuse-activation-memory-allocations",
-    llvm::cl::desc("Should activation memory allocations be reused"),
+    llvm::cl::desc("Reuse activations memory between different IR functions"),
```

Comment:
Here you repurpose the flag for something different. 

The original intent of this flag was to prohibit reuse (at the expense of using more memory), so that the memory for each activation is never used for anything else. It is useful for debugging on e.g. new HW/bare metal, where you want to check post-mortem, after you dumped all of the memory used jointly by activations, that memory segments for specific activations have specific values.

Reply:
The meaning of this flag has indeed changed from its previous use (cannot do otherwise). Previously this flag enabled us to allocate exclusive activations buffers within functions and between functions. Now it can only enable us to allocate exclusive activations between functions but not within functions (the buffers within same function will always be allocated in a compact-optimized way). But you said that you don't mind if I break it. Instead I repurposed it for something similar.